app/src/test/java/ch/hikemate/app/model/facilities/FacilitiesViewModelTest.kt

```diff
@@ -0,0 +1,183 @@
+package ch.hikemate.app.model.facilities
+
+import ch.hikemate.app.model.route.Bounds
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.TestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.osmdroid.util.GeoPoint
+
+class FacilitiesViewModelTest {
+
+  private lateinit var mockFacilitiesRepository: FacilitiesRepository
+  private lateinit var testDispatcher: TestDispatcher
+
+  private lateinit var facilitiesViewModel: FacilitiesViewModel // SUT
+
+  private val testBounds = Bounds(46.5, 6.6, 46.51, 6.62)
+  private val testFacilities = listOf(Facility(FacilityType.TOILETS, GeoPoint(46.51, 6.61)))
+
+  @OptIn(ExperimentalCoroutinesApi::class)
+  @Before
+  fun setUp() {
+    testDispatcher = StandardTestDispatcher()
+    Dispatchers.setMain(testDispatcher)
+
+    mockFacilitiesRepository = mock()
+    facilitiesViewModel = FacilitiesViewModel(mockFacilitiesRepository, testDispatcher)
+  }
+
+  @OptIn(ExperimentalCoroutinesApi::class)
+  @After
+  fun tearDown() {
+    Dispatchers.resetMain()
+  }
+
+  @Test
+  fun testGetFacilities_onSuccess() {
+    val facilities = listOf(Facility(FacilityType.TOILETS, GeoPoint(46.51, 6.61)))
+
+    `when`(mockFacilitiesRepository.getFacilities(any(), any(), any())).then {
+      val onSuccess = it.getArgument<(List<Facility>) -> Unit>(1)
+      onSuccess(facilities)
+    }
+
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { result -> assertEquals(facilities, result) },
+        onFailure = { fail("Should not be called") })
+  }
+
+  @Test
+  fun testGetFacilities_onFailure() {
+    val exception = Exception("Test exception")
+
+    `when`(mockFacilitiesRepository.getFacilities(any(), any(), any())).then {
+      val onFailure = it.getArgument<(Exception) -> Unit>(2)
+      onFailure(exception)
+    }
+
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { fail("Should not be called") },
+        onFailure = { ex -> assertEquals(exception, ex) })
```

Comment:
Warning: same remark as above. Please check the the callback is called with a boolean